feat: Add support for MinIO storage in rclonefs

frin1 opened the pull request:
This commit introduces support for MinIO storage in the rclonefs package. It adds a new case in the switch statement in the NewFromConfig function to handle the "minio" disk type.

ngocphuongnb commented:
@frin1 Thank you so much, I appreciate your contribution!

I have some concerns as follows:
Minio itself is an S3-compatible storage solution, so we can use `rclone backend s3` to work with Minio.
In the past, I tested `rclonefs` with Minio and it worked, just add the following config to the `STORAGE_DISKS` env:
```
[
 {
 "name": "my_minio",
 "driver": "s3",
 "root": "/files",
 "provider": "Minio",
 "endpoint": "http://localhost:9000",
 "region": "",
 "bucket": "fastschematest",
 "access_key_id": "access_key_id",
 "secret_access_key": "secret_access_key",
 "base_url": "https://cdn.site.local"
 }
]
```
I just tested it again and found that we need to set `force_path_style=true` for `rclonefs` to be able to work with Minio:
```
//file: pkg/rclonefs/s3.go:61
cfgMap.Set("force_path_style", "true")
```
Hence I found that we can directly use rclone instead of `github.com/minio/minio-go/v7`
The `miniofs` methods are very useful, can you keep them and move them to s3.go?
- PresignedURL
- URL
- GetURL

In addition, the new file `fs/diskbase` is very useful, I think it is reasonable to add it.

If you don't have enough time, can I create a commit to update?

frin1 commented:
@ngocphuongnb It makes sense to stick with the rclone backend for using MinIO as a backend. I tested with `cfgMap.Set("force_path_style", "true")`, and it worked.

I have created a new commit with `cfgMap.Set("force_path_style", "true")` when selecting Minio as the provider. The changes I made based on your suggestions are as follows:

- Changed the `URL` function to use similar logic as the `miniofs` package was using.
- Refactored `rclonefs` to use `fs.DiskBase` (pulled out the functions from `BaseRcloneDisk`), as this can be used by others if an integration is used later that is not supported by `rclone`.
- Updated the unit tests.

I did not keep `GetURL` and `PresignedURL` because `GetURL` is not a part of the `fs.Disk` interface, and `PresignedURL` is probably not that useful since it only keeps the link available for a certain time. At least for my use case, which is a blog, I don't want my images to become unavailable due to an expired URL. However, feel free to add `PresignedURL` if you think it would be useful.

ngocphuongnb commented:
LGTM
